crates/matrix-sdk/src/sliding_sync.rs

```diff
@@ -1413,50 +1415,41 @@ impl SlidingSyncViewRequestGenerator {
 
         trace!(end, max_index, range_end, name = self.view.name, "updating state");
 
-        if let InnerSlidingSyncViewRequestGenerator::PagingFullSync { position: _, limit, .. }
-        | InnerSlidingSyncViewRequestGenerator::GrowingFullSync {
-            position: _, limit, ..
-        } = self.inner
-        {
-            let max = limit.unwrap_or(max_index);
-            if end >= max {
-                trace!(name = self.view.name, "going live");
-                // we are switching to live mode
-                self.view.set_range(0, max);
-                self.inner = InnerSlidingSyncViewRequestGenerator::Live
-            } else {
-                self.view.set_range(0, end);
-            }
-        }
-
         match self.inner {
             InnerSlidingSyncViewRequestGenerator::PagingFullSync {
-                position: _,
-                batch_size,
+                ref mut position,
+                ref mut live,
```

Comment:
Can we do a `match &mut self.inner` instead and dereference things like `limit` where necessary? I'd prefer if we consistently used the match ergonomics style of obtaining references in `match`, rather than `ref` (`mut`).